### ISSUE TITLE
Increment sliders can't go to the last value if touched in settings menu

### ISSUE THREAD

**Issue by epicEaston197** [Yellow-Dog-Man/Resonite-Issues#2095]
### Describe the Underlying Issue

Settings like per pixel lights can't go beyond 15 if they're touched because the slider is incapable of going to 16 but you could still manually type 16 into the text box if needed

### To Reproduce

1. Modify a setting that has an increment slider for example per pixel lights in the graphic settings
2. You'll notice that you can't put the slider to 16 or the 2nd to Max value

### Expected Behavior

The slider should be able to go from zero to Max

### Screenshots / Video

https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76523692/3d6e4dc4-6ce1-4a4e-af25-62f7d4a52912

### Reporters

@epicEaston197

**Comment by shiftyscales** [comment]
I had noticed this as well on the anisotropic filtering slider- expectation would be that you should be able to set 16 samples- as it is a fairly common value.

**Comment by epicEaston197** [comment]
> I had noticed this as well on the anisotropic filtering slider- expectation would be that you should be able to set 16 samples- as it is a fairly common value.

Yes the per pixel lights was an example the issue is about increment sliders that includes the anisotropic level because it too is an increment slider

**Comment by RyuviTheViali** [comment]
This is actually an issue specific to sliders of integer types. Sliders of type int will only allow slider movement between the slider's minimum and it's maximum exclusive (min -> max-1) instead of max inclusive. This can be worked around by just adding 1 to the max, but a more direct solution is to have the maximum be set to inclusive, or have it be an option on sliders.

**Comment by shiftyscales** [comment]
I see. Thanks for looking into it @RyuviTheViali.

Seeking feedback from @Frooxius on the best path forward here.

**Comment by Banane9** [comment]
> This is actually an issue specific to sliders of integer types. Sliders of type int will only allow slider movement between the slider's minimum and it's maximum exclusive (min -> max-1) instead of max inclusive.

I bet this is because of #1530

**Comment by shiftyscales** [comment]
I believe this issue was worked around by @RyuviTheViali - as currently it is possible to set the anisotropy slider to 16.
